File: 345-ReverseVowelsOfAString.java
public class Solution {
    public String reverseVowels(String s) {
        if (s.length() == 0 || s == null){
            return s;
        }
        String vowels = "aeiouAEIOU";
        int start = 0;
        int end = s.length() - 1;
        char[] chars = s.toCharArray();

        while (start < end){
            while (!vowels.contains(chars[start]+"") && start < end){
                start ++;
            }
            while (!vowels.contains(chars[end]+"") && start < end){
                end --;
            }
            
            char tmp = chars[start];
            chars[start] = chars[end];
            chars[end] = tmp;
            start ++;
            end --;
        }
        return new String(chars);
    }
}

File: cpp/345-ReverseVowelsOfaString.cpp
class Solution {
public:
    string reverseVowels(string s) {
        int start = 0, end = s.length() - 1;
        unordered_set<char> vows = {'a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U'};
        while (start < end){
            if (!vows.count(s[start])){
                start++;
            }else if (!vows.count(s[end])){
                end--;
            }else{
                swap(s[start], s[end]);
                start++;
                end--;
            }
        }
        return s;
    }
};